### DISCUSSION TITLE
size property should be optional in FileMessage?

### DISCUSSION THREAD

**Discussion by gtu-myowin** [flyerhq/flutter_chat_ui#650] [Ideas & Requests]
Hello, this is just a small feedback. I think the `size` property on `FileMessage` type should be optional instead of required because the file message is working fine even if I pass just zero. 

Yes, it shows 0KB under the file name but since it is working without it anyway, wouldn't it be better to set it as optional and only show the size if it is being passed?

**Comment by demchenkoalex** [comment]
Hi! Did it as required because I noticed pretty much all the time libraries I used to pick files returned size as well, and it is nicer when you see exactly how large this thing is before trying to open it. Do you have a use case when you can't access size?

**Comment by gtu-myowin** [comment]
Thanks for your reply. Yes, it is nicer to show the file size but like you made `showUserAvatars` option for the `Chat()` widget, it would be nicer to make the `size` property optional as well.

I am using Laravel Chatify package for the backend which uses Pusher Channels for the real time message delivering. That package doesn't contain the file size by default and my customization isn't working either. I have asked the package author about that here https://github.com/munafio/chatify/issues/200 and still waiting for his response.

But the main reason, I posted here as well is as a suggestion. I am like, "if it is working without the `size` property, why it is required?".

**Comment by gtu-myowin** [comment]
Since, this is just a suggestion. I will just close this. Thank you very much for your support bro.

**Comment by demchenkoalex** [comment]
No need to close, I will forget about it otherwise. I will add this.

**Comment by gtu-myowin** [comment]
Thank you very much 😁
